src/app/firebase.service.ts

```diff
@@ -1,13 +1,43 @@
 import { Injectable } from '@angular/core';
-import { AngularFirestore, AngularFirestoreDocument } from '@angular/fire/firestore';
+import { AngularFirestore } from '@angular/fire/firestore';
 import { Observable } from 'rxjs';
+import { Session, SessionWithId } from './session';
+import { map } from 'rxjs/operators';
+import { FirebaseRound } from './round';
 
 @Injectable({
   providedIn: 'root'
 })
 export class FirebaseService {
+  allSessions$: Observable<any[]> = this.firestore.collection('sessions').valueChanges();
 
   constructor(public firestore: AngularFirestore) { }
 
-  allSessions$: Observable<any[]> = this.firestore.collection('sessions').valueChanges();
+  /**
+   * Return an observable stream of the session data for the session with
+   * Firebase ID `id`.
+   *
+   * (In case you need the Firebase ID for later, it's saved as a property on
+   * the SessionWithId objects emitted from the observable.)
+   */
+  public getSession(id: string): Observable<SessionWithId> {
+    return this.firestore
+      .collection('sessions')
+      .doc<Session>(id)
+      .snapshotChanges()
+      .pipe(map(action => ({id: action.payload.id, ...action.payload.data()})));
+  }
+
+  /**
+   * Return an observable stream of the round data for the round whose
+   * Firebase ID is `roundId` within the session denoted by `sessionId`.
+   */
+  public getRound(roundPath: {sessionId: string, roundId: string}): Observable<FirebaseRound> {
```

Comment:
We should probably make this just take `sessionId` and `roundId` as separate parameters, if we have them as an object we can always use `...roundPath`.

Reply:
You know, I tried that first, but the `...roundPath` doesn't work—JavaScript isn't smart enough to associate object keys with parameter names :-(

See also: https://stackoverflow.com/questions/48008539/how-to-spread-an-object-to-a-function-as-arguments

Reply:
Ah ok. Nevermind then.